check.py

```diff
@@ -75,14 +79,55 @@ def fail_if_not_contained(actual, expected):
 else:
   tests = requested[:]
 
+warnings = []
+
 if not interpreter:
-  print 'warning: no interpreter provided (not testing spec interpreter validation)'
+  warnings.append('warning: no interpreter provided (did not test spec interpreter validation)')
 if not has_node:
-  print 'warning: no node found (not checking proper js form)'
+  warnings.append( 'warning: no node found (did not check proper js form)')
 if not has_mozjs:
-  print 'warning: no mozjs found (not checking asm.js validation)'
+  warnings.append( 'warning: no mozjs found (did not check asm.js validation)')
 if not has_emcc:
-  print 'warning: no emcc found (not checking emscripten/binaryen integration)'
+  warnings.append('warning: no emcc found (did not check non-vanilla emscripten/binaryen integration)')
+
+# setup
+
+WATERFALL_BUILD = os.path.join('test', 'waterfall_build')
+
+def fetch_waterfall():
+  rev = open(os.path.join('test', 'revision')).read()
+  try:
+    local_rev = open(os.path.join('test', 'local-revision')).read()
+  except:
+    local_rev = None
+  if local_rev == rev: return
```

Comment:
I'd instead just do `os.path.exists('wasm-binaries-%s.tbz2' % rev)`, no need for `local_rev`. That's what `update.py` does. Could you use #117's code for that?